### PR TITLE
Updated action to support adding optional environment variables and secrets 

### PR DESCRIPTION
**Pull request by pavangudiwada**
Sample
image

### CODE REVIEW

File: prometheus_remediation/job_runner.py [hunk 1]
```diff
@@ -85,6 +68,31 @@ class JobParams(ActionParams):
     completion_timeout: int = 300
     backoff_limit: int = None  # type: ignore
     active_deadline_seconds: int = None  # type: ignore
+    env_var: Optional[List[EnvVar]] = None
```
Comment (aantn) [inline prometheus_remediation/job_runner.py]: Should be `env_vars` (plural)

File: prometheus_remediation/job_runner.py [hunk 2]
```diff
@@ -137,12 +145,16 @@ def run_job_from_alert(event: PrometheusKubernetesAlert, params: JobParams):
...
-            FileBlock("job-runner-logs.txt", pod.get_logs())
-            ])
+        try:
+            wait_until_job_complete(job, params.completion_timeout)
+            job = hikaru.from_dict(job.to_dict(), cls=RobustaJob)  # temporary workaround for https://github.com/haxsaw/hikaru/issues/15
+            pod = job.get_single_pod()
+            event.add_enrichment([
+                FileBlock("job-runner-logs.txt", pod.get_logs())
+                ])
+        except Exception as e:
```
Comment (aantn) [inline prometheus_remediation/job_runner.py:160]: You might swallow other exceptions (not just the one you are looking for)
Verify it is the right exception and rethrow it if it isn't the exception you are looking for.

See an example of how to do this here - https://stackoverflow.com/a/13531310/495995

File: prometheus_remediation/job_runner.py [hunk 3]
```diff
@@ -137,12 +145,16 @@ def run_job_from_alert(event: PrometheusKubernetesAlert, params: JobParams):
...
+            print(f"Timed out running {job_name} for alert {event.alert_name}")
```
Comment (aantn) [inline prometheus_remediation/job_runner.py]: Use `logging.warning` not `print`.

Also, you're duplicating the string here and below. Make one copy, then log it and add it to the markdown block.

E.g:

```
err = f"...."
logging.warning(err)
event.add_enrichment([MarkdownBlock(err)])
```

As they say, DRY (don't repeat yourself)

**Review by aantn — CHANGES_REQUESTED** [review]
Very nice. Some comments and requests for small fixes.

**Comment by pavangudiwada** [comment]
New alert
image

File: prometheus_remediation/job_runner.py [hunk 4]
```diff
@@ -136,13 +144,29 @@ def run_job_from_alert(event: PrometheusKubernetesAlert, params: JobParams):
 
     job.create()
 
+
+    info_messages = []
     if params.notify:
-        event.add_enrichment([MarkdownBlock(f"Created job from alert: *{job_name}*.")])
+        info_messages.append(f"Created Job from alert: *{job_name}*.")
```
Comment (aantn) [inline prometheus_remediation/job_runner.py]: Can you make the "Create Job from alert" bold and the job itself not bold. (That will better match the style in the rest of the Slack message.)

File: prometheus_remediation/job_runner.py [hunk 5]
```diff
@@ -136,13 +144,29 @@ def run_job_from_alert(event: PrometheusKubernetesAlert, params: JobParams):
...
+        try:
+            wait_until_job_complete(job, params.completion_timeout)
+            job = hikaru.from_dict(job.to_dict(), cls=RobustaJob)  # temporary workaround for https://github.com/haxsaw/hikaru/issues/15
+            pod = job.get_single_pod()
+            event.add_enrichment([
+                FileBlock("job-runner-logs.txt", pod.get_logs())
+                ])
+        except Exception as e:
+            if str(e) != "Failed to reach wait condition":
+                warning_msg = f"Action Job stopped due to Exception {e}"
```
Comment (aantn) [inline prometheus_remediation/job_runner.py]: Can you rephrase as "Error running Job: {e}"

File: prometheus_remediation/job_runner.py [hunk 6]
```diff
@@ -136,13 +144,29 @@ def run_job_from_alert(event: PrometheusKubernetesAlert, params: JobParams):
...
+                logging.warning(warning_msg)
+                info_messages.append(warning_msg)
+            else:
+                err_str = f"Status: Timed out, could not fetch output."
+                logging.warning(f"Action Job {job_name} - {err_str}")
```
Comment (aantn) [inline prometheus_remediation/job_runner.py]: Can you rephrase as "Failed to fetch Job result: {err_str}"

**Review by aantn — CHANGES_REQUESTED** [review]
Left some minor comments. Also please update the readme.

Finally, can you share details on how you tested this? (E.g. did you test the case where the job succeeds + the case where it fails + the case where it times out?)

File: README.md [hunk 7]
```diff
@@ -20,16 +20,23 @@ customPlaybooks:
 # Change the following line according to your needs
 - triggers:
   - on_prometheus_alert:
-      alert_name: TestAlert
+      alert_name: KubeJobFailed
```
Comment (aantn) [inline README.md]: Can we use an alert that is *not* job-related. It's a little confusing that we're running a Job as a result but also triggering this on a failed job. Best to change the trigger and keep everything clear with one use of jobs here (the real one)

**Review by aantn — APPROVED** [review]
Approved, with one request for change.
